<?xml version="1.0"?>	
<robot name="robot">

  <material name="Orange">
    <color rgba="0.5 0.2 0.0 1" />
  </material>
  <material name="Yellow">
    <color rgba="0.5 0.5 0.0 1" />
  </material>
  <material name="Black">
    <color rgba="0.5 0.0 0.0 1" />
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.4 1" />
  </material>
  <material name="Green">
    <color rgba="0.0 0.4 0.0 1" />
  </material>
  <material name="White">
    <color rgba="1 1 1 1" />
  </material>

  <link name="base_footprint" />

  <link name="base_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="70.0" />
      <inertia ixx="0.079945" ixy="0.00012078" ixz="0.00015606" iyy="0.068406" iyz="-0.0049053" izz="0.12343" />
    </inertial>
  <visual>
    <origin xyz="0 0 -0.015" rpy="0 0 -1.57" />
    <geometry>
      <mesh filename="package://mobile_robot/urdf/meshes/Base_Plate.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="Blue" />
  </visual>
  <collision>
    <origin xyz="0 0 -0.015" rpy="0 0 -1.57" />
    <geometry>
      <mesh filename="package://mobile_robot/urdf/meshes/Base_Plate.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
  </link>


  <joint name="base_joint" type="fixed">
    <origin xyz="0 0 0.05335" rpy="0 0 0" />
    <parent link="base_footprint" />
    <child link="base_link" />
    <axis xyz="0 0 0" />
  </joint>

  <link name="robot_body_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="70.0" />
      <inertia ixx="0.079945" ixy="0.00012078" ixz="0.00015606" iyy="0.068406" iyz="-0.0049053" izz="0.12343" />
    </inertial>
  <visual>
    <origin xyz="0.0 0 -0.19125" rpy="0 0 -1.57" />
    <geometry>
      <mesh filename="package://mobile_robot/urdf/meshes/Robot_Cover_Full.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="Yellow" />
  </visual>
  <collision>
    <origin xyz="0.0 0 -0.19125" rpy="0 0 -1.57" />
    <geometry>
      <mesh filename="package://mobile_robot/urdf/meshes/Robot_Cover_Full.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
  </link>

  <joint name="robot_body_joint" type="fixed">
    <origin xyz="0 0 0.175" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="robot_body_link" />
    <axis xyz="0 0 0" />
  </joint>

  <link name="right_wheel_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
      <mass value="0.50917" />
      <inertia ixx="0.0012185" ixy="1.8059E-20" ixz="3.0427E-20" iyy="0.0012185" iyz="-4.0491E-20" izz="0.0022928" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder length="0.03" radius="0.04" />
      </geometry>
      <material name="Black" />
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder length="0.03" radius="0.04" />
      </geometry>
    </collision>
  </link>

  <joint name="right_wheel_joint" type="continuous">
    <origin xyz="0.0 -0.12 0.01" rpy="1.571 0 0" />
    <parent link="base_link" />
    <child link="right_wheel_link" />
    <axis xyz="0 0 -1" />
  </joint>

  <link name="left_wheel_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
      <mass value="0.50917" />
      <inertia ixx="0.0012185" ixy="1.8059E-20" ixz="3.0427E-20" iyy="0.0012185" iyz="-4.0491E-20" izz="0.0022928" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder length="0.03" radius="0.04" />
      </geometry>
      <material name="Black" />
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder length="0.03" radius="0.04" />
      </geometry>
    </collision>
  </link>

  <joint name="left_wheel_joint" type="continuous">
    <origin xyz="0.0 0.12 0.01" rpy="-1.571 0 0" />
    <parent link="base_link" />
    <child link="left_wheel_link" />
    <axis xyz="0 0 1" />
  </joint>

  <link name="front_castor_wheel_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
      <mass value="0.50917" />
      <inertia ixx="0.0012185" ixy="1.8059E-20" ixz="3.0427E-20" iyy="0.0012185" iyz="-4.0491E-20" izz="0.0022928" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.03" />
      </geometry>
      <material name="Black" />
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.03" />
      </geometry>
    </collision>
  </link>

  <joint name="front_castor_wheel_joint" type="fixed">
    <origin xyz="0.07 0.0 0.0" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="front_castor_wheel_link" />
    <axis xyz="0 1 1" />
  </joint>

  <link name="rear_castor_wheel_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
      <mass value="0.50917" />
      <inertia ixx="0.0012185" ixy="1.8059E-20" ixz="3.0427E-20" iyy="0.0012185" iyz="-4.0491E-20" izz="0.0022928" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.03" />
      </geometry>
      <material name="Black" />
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <sphere radius="0.03" />
      </geometry>
    </collision>
  </link>

  <joint name="rear_castor_wheel_joint" type="fixed">
    <origin xyz="-0.07 0.0 0.0" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="rear_castor_wheel_link" />
    <axis xyz="0 1 1" />
  </joint>


  <link name="lidar_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.0093378" />
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0" /> 
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder length="0.046" radius="0.035" />
      </geometry>
      <material name="Green" />
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder length="0.046" radius="0.035" />
      </geometry>
    </collision>
  </link>

  <joint name="laser_joint" type="fixed">
    <origin xyz="0.0075 0.0 0.35" rpy="0 0 3.14" />
    <parent link="base_link" />
    <child link="lidar_link" />
    <axis xyz="0 0 0" />
  </joint>





  <!-- Transmission (for ros_control) -->
  <transmission name="left_wheel_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_wheel_motor"><mechanicalReduction>1</mechanicalReduction></actuator>
  </transmission>

  <transmission name="right_wheel_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="right_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_wheel_motor"><mechanicalReduction>1</mechanicalReduction></actuator>
  </transmission>


</robot>
